src/main/scala/mirroring/services/writer/SchemaNotMatchException.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (2021) The Delta Flow Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mirroring.services.writer
+
+/** Exception thrown when a schema does not match for source and destination.
+  */
+private[writer] class SchemaNotMatchException() extends Exception
```

Comment:
maybe let's move this class to an `exceptions` package?

Reply:
I got this idea from [spark](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/TaskNotSerializableException.scala) library. This approach has two benefits:
  - exception remains private to `writer`
  - exception needs no import statement inside `mirroring.services.writer`